### PR TITLE
Besides raw text string, tokens can be passed to the 'generate' method

### PR DESCRIPTION
**Pull request by xiaohan2012**
As far as I understood, custom tokenization is not easy to use for the current version. This pull request allows custom tokenization to be used.

### CODE REVIEW

File: wordcloud/wordcloud.py [hunk 1]
```diff
@@ -233,8 +233,8 @@ def fit_words(self, words):
         self.layout_ = list(zip(words, font_sizes, positions, orientations, colors))
         return self.layout_
 
-    def process_text(self, text):
-        """Splits a long text into words, eliminates the stopwords.
+    def _get_words(self, text):
```
Comment (amueller) [inline wordcloud/wordcloud.py]: Why did you remove this public function? Backward incompatible changes make it hard for others to use the code.

File: wordcloud/wordcloud.py [hunk 2]
```diff
@@ -300,8 +318,15 @@ def process_text(self, text):
 
         return words
 
-    def generate(self, text):
-        """Generate wordcloud from text.
+    def generate(self, stuff):
```
Comment (amueller) [inline wordcloud/wordcloud.py]: Stuff is not a good parameter name.

Comment (amueller) [inline wordcloud/wordcloud.py]: I would rather have different functions for different input formats, like ``generate_from_tokens`` or something like this. It is also not really clear from the docstring what input format is expected.

**Comment by amueller** [comment]
Your code is not python3 compatible (see travis).

File: wordcloud/wordcloud.py (hunk 1 above)
Comment (xiaohan2012) [inline wordcloud/wordcloud.py]: Thanks for pointing out. I don't know that until you told me, as I am only modifying to my own need.

File: wordcloud/wordcloud.py (hunk 2 above)
Comment (xiaohan2012) [inline wordcloud/wordcloud.py]: Good to know that!

**Comment by amueller** [comment]
Sorry for the slow reply. So is the idea here that you could overload specific methods to get custom tokenization?
I think I would rather pass a callable tokenizer to the constructor (or maybe to the function?).
Also, where do you use isstr?

**Comment by amueller** [comment]
this can be achieved using ``generate_from_frequencies``. also, no replies :-/
